plugins/services/src/js/utils/ServiceUtil.js

```diff
@@ -135,27 +108,59 @@ const ServiceUtil = {
     return serviceName[serviceName.length - 1];
   },
 
-  convertServiceLabelsToArray(service) {
-    if (!(service instanceof Service)) {
-      return [];
-    }
-
-    const labels = service.getLabels();
-    if (labels == null) {
-      return [];
-    }
-
-    return Object.keys(labels).map(key => ({ key, value: labels[key] }));
+  isSDKService(service) {
+    return (
+      service &&
+      service.getLabels &&
+      typeof service.getLabels === "function" &&
+      service.getLabels().DCOS_COMMONS_API_VERSION != null
+    );
   },
 
-  isSDKService(service) {
-    if (!(service instanceof Service)) {
-      return false;
+  /**
+   * this function generates webui url from given labels.
+   *
+   * - DCOS_SERVICE_WEB_PATH is a newly introduced label which
+   *   contains a url part starting with `/`
+   * - DCOS_SERVICE_PORT_INDEX & DCOS_SERVICE_SCHEME are legacy
+   *   labels used to detexct if service has webui
+   * - legacy labels were disabled for SDK Packages (detected via
+   *   DCOS_COMMONS_API_VERSION label)
+   *
+   * @param {object} labels object from service
+   * @param {string} rootUrl from config
+   * @returns {string} url or empty string
+   */
+  getWebURL(labels, rootUrl) {
+    const {
+      DCOS_SERVICE_NAME,
+      DCOS_SERVICE_PORT_INDEX,
+      DCOS_SERVICE_SCHEME,
+      DCOS_SERVICE_WEB_PATH,
+      DCOS_COMMONS_API_VERSION
+    } = labels;
+    const serviceName = encodeURIComponent(DCOS_SERVICE_NAME);
+    const baseUrl = `${rootUrl}/service/${serviceName}`;
+
+    if (serviceName && DCOS_SERVICE_WEB_PATH) {
+      const webPath =
+        DCOS_SERVICE_WEB_PATH.charAt(0) === "/"
+          ? DCOS_SERVICE_WEB_PATH
+          : "/" + DCOS_SERVICE_WEB_PATH;
```

Comment:
as i played around with the code before, i have a patch here. i like it a little better. proceed as you like.

```diff
diff --git a/plugins/services/src/js/utils/ServiceUtil.js b/plugins/services/src/js/utils/ServiceUtil.js
index f6515b0d0..c040c758c 100644
--- a/plugins/services/src/js/utils/ServiceUtil.js
+++ b/plugins/services/src/js/utils/ServiceUtil.js
@@ -133,34 +133,23 @@ const ServiceUtil = {
    */
   getWebURL(labels, rootUrl) {
     const {
-      DCOS_SERVICE_NAME,
-      DCOS_SERVICE_PORT_INDEX,
-      DCOS_SERVICE_SCHEME,
-      DCOS_SERVICE_WEB_PATH,
-      DCOS_COMMONS_API_VERSION
+      DCOS_COMMONS_API_VERSION: apiVersion,
+      DCOS_SERVICE_NAME: name,
+      DCOS_SERVICE_PORT_INDEX: portIndex,
+      DCOS_SERVICE_SCHEME: scheme,
+      DCOS_SERVICE_WEB_PATH: webPath = ""
     } = labels;
-    const serviceName = encodeURIComponent(DCOS_SERVICE_NAME);
-    const baseUrl = `${rootUrl}/service/${serviceName}`;
+    const serviceName = encodeURIComponent(name);
+    const hasNewWebUiLabel = serviceName && webPath;
+    const hasOldWebUiLabel = serviceName && portIndex && scheme && !apiVersion;

-    if (serviceName && DCOS_SERVICE_WEB_PATH) {
-      const webPath =
-        DCOS_SERVICE_WEB_PATH.charAt(0) === "/"
-          ? DCOS_SERVICE_WEB_PATH
-          : "/" + DCOS_SERVICE_WEB_PATH;
-
-      return `${baseUrl}${webPath}`;
-    }
-
-    if (
-      serviceName &&
-      !DCOS_COMMONS_API_VERSION &&
-      DCOS_SERVICE_PORT_INDEX &&
-      DCOS_SERVICE_SCHEME
-    ) {
-      return `${baseUrl}/`;
+    if (!hasNewWebUiLabel && !hasOldWebUiLabel) {
+      return "";
     }

-    return "";
+    return `${rootUrl}/service/${serviceName}/${
+      webPath.startsWith("/") ? webPath.substring(1) : webPath
+    }`;
   }
 };
```

Reply:
yup, yours is better, mind just commiting to the branch? seems like its on your machine anyway :)

Reply:
Pushed. What do you think on slashing slashes? 

> Also i don't feel confident about manipulating slashes and rather anticipated a "of course there are always slashes, as the API-docs show here". Now that this is not the case i'm not sure whether it's better in or out.

Also someone yesterday forbid me to approve this PR now. :)

Reply:
* slashes: its just a string and there is absolutely no enforcement that guarantees it starting with a slash - I'd keep the check here (even if I generally agree with you)
* approve: I reviewed your code - you can approve it again ("in my behalf" when it comes to your commit)